airbyte-webapp/src/views/Connector/ConnectorForm/useBuildForm.tsx

```diff
@@ -13,21 +14,35 @@ import { buildYupFormForJsonSchema } from "core/jsonSchema/schemaToYup";
 
 import { ConnectorFormValues } from "./types";
 
-export function useBuildInitialSchema(
-  connectorSpecification?: ConnectorDefinitionSpecification
-): JSONSchema7Definition | undefined {
-  return useMemo(() => {
-    return connectorSpecification?.connectionSpecification as JSONSchema7Definition | undefined;
-  }, [connectorSpecification]);
-}
-
-// useBuildForm hook
 export interface BuildFormHook {
   initialValues: ConnectorFormValues;
   formFields: FormBlock;
+  jsonSchema: JSONSchema7;
 }
 
-export function useBuildForm(jsonSchema: JSONSchema7, initialValues?: Partial<ConnectorFormValues>): BuildFormHook {
+export function useBuildForm(
+  formType: "source" | "destination",
+  selectedConnectorDefinitionSpecification: ConnectorDefinitionSpecification,
+  initialValues?: Partial<ConnectorFormValues>
+): BuildFormHook {
+  const { formatMessage } = useIntl();
+
+  const jsonSchema: JSONSchema7 = useMemo(
+    () => ({
+      type: "object",
+      properties: {
+        name: {
+          type: "string",
+          title: formatMessage({ id: `form.${formType}Name` }),
+          description: formatMessage({ id: `form.${formType}Name.message` }),
+        },
+        connectionConfiguration:
+          selectedConnectorDefinitionSpecification.connectionSpecification as JSONSchema7Definition,
+      },
+      required: ["name"],
+    }),
+    [formType, formatMessage, selectedConnectorDefinitionSpecification.connectionSpecification]
+  );
```

Comment:
One potential future issue with putting this in the `useBuildForm` hook is that when we want to use the connector form in the Connector Builder for the config menu, we don't want to have this "connector name" field in there.

Can definitely be addressed in a later PR but I wanted to call it out since I thought of it